docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+        placeholder="Male",
+        name="gender",
+    ),
+)
+```
+
+This form is all very compact as you can see from the example, so we need to add some styling to make it look better. We can do this by adding a `vstack` component around the form fields. The `vstack` component stacks the form fields vertically. Check out the [layout] docs !!!! for more information on how to layout your app.
+
+
+```python demo
+rx.form(
+    rx.vstack(
+        rx.input(
+            placeholder="User Name", name="name", required=True
+        ),
+        rx.input(
+            placeholder="user@reflex.dev",
+            name="email",
+        ),
+        rx.select(
+            ["Male", "Female"],
+            placeholder="Male",
+            name="gender",
+        ),
+    ),
+)
+```
+
+Now you have probably realised that we have all the form fields, but we have no way to submit the form. We can add a submit button to the form by adding a `rx.button` component to the `vstack` component. The `rx.button` component takes in the text that is displayed on the button and the `type` prop which is the type of button. The `type` prop is set to `submit` so that the form is submitted when the button is clicked. 
+
+In addition to this we need a way to update the `users` state variable when the form is submitted. All state changes are handled through functions in the state class, called [event handlers]({docs.events.events_overview.path}).
+
+Components have special props called event triggers, such as `on_submit`, that can be used to make components interactive. Event triggers connect components to event handlers, which update the state.
+
+The `on_submit` event trigger of `rx.form` is hooked up to the `add_user` event handler that is defined in the `State` class. The `add_user` event handler takes in the form data as a dictionary and appends it to the `users` state variable.
+
+
+```python
+class State(rx.State):
+    
+    ...
+
+    def add_user(self, form_data: dict):
+        self.users.append(User(**form_data))
+
+
+def form():
+    return rx.form(
+        rx.vstack(
+            rx.input(
+                placeholder="User Name", name="name", required=True
+            ),
+            rx.input(
+                placeholder="user@reflex.dev",
+                name="email",
+            ),
+            rx.select(
+                ["Male", "Female"],
+                placeholder="Male",
+                name="gender",
+            ),
+            rx.button("Submit", type="submit"),
+        ),
+        on_submit=State.add_user,
+        reset_on_submit=True,
+    )
+```
+
+Finally we must add the new `form()` component we have defined to the `index()` function so that the form is rendered on the page.
+
+Below is the full code for the app so far. If you try this form out you will see that you can add new users to the table by filling out the form and clicking the submit button. The form data will also appear as a toast on the screen when submitted.
```

Comment:
What is a toast? Maybe say it will appear "as a small window in the corner" or something.

Reply:
added